nixos/tests/nextcloud/basic.nix

```diff
@@ -0,0 +1,61 @@
+import ../make-test.nix ({ pkgs, ...} : {
+  name = "nextcloud-basic";
+  meta = with pkgs.stdenv.lib.maintainers; {
+    maintainers = [ globin eqyiel ];
+  };
+
+  nodes = {
+    client = { config, pkgs, ... }: {
+      services.davfs2.enable = true;
+
+      environment.etc."davfs2/secrets" = {
+        mode = "600";
+        text = ''
+          http://nextcloud/remote.php/dav/files/root root notproduction
+        '';
+      };
+    };
+
+    nextcloud = { config, pkgs, ... }: {
+      networking.firewall.allowedTCPPorts = [ 80 ];
+
+      services.nextcloud = {
+        enable = true;
+        nginx.enable = true;
+        hostName = "nextcloud";
+        config.adminpass = "notproduction";
+      };
+
+      services.davfs2.enable = true;
+
+      environment.etc."davfs2/secrets" = {
+        mode = "600";
+        text = ''
+          http://nextcloud/remote.php/dav/files/root root notproduction
+        '';
+      };
+    };
+
+  };
+
+  testScript = let
+    diffSharedFile = pkgs.writeScript "diff-shared-file" ''
+      #!${pkgs.stdenv.shell}
+      echo "waiting for FUSE..."
+      sleep 10
```

Comment:
cc @xeji 

Could you do something like:

```
while ! mountpoint -q /mnt; do
  sleep 0.1
end
```

I am pretty sure 10 seconds will be not enough at some point under heavy load.

Reply:
Or does it wait for davfs to fetch results?

Reply:
Just fetching a file with rclone might be good enough as a test: `https://rclone.org/webdav/`

Reply:
@Mic92 that sounds like a better solution, I'll make it happen.